serve_test.ts

```diff
@@ -79,7 +97,7 @@ for (const tc of testCases) {
       const reqInit: RequestInit = { method: tc.method };
       if (typeof tc.params === 'string') {
         reqInit.body = tc.params;
-        reqInit.headers = { 'content-type': 'application/json' };
+        reqInit.headers = Object.assign({ 'content-type': 'application/json' }, tc.headers);
```

Comment:
In this case, I prefer this form.
```suggestion
        reqInit.headers = tc.headers || { 'content-type': 'application/json' };
```